src/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { isObject } from '@metamask/utils';
+
+export const DEDICATED_WORKER_NAME = 'dedicatedWorker';
+
+export type StreamData = number | string | Record<string, unknown> | unknown[];
```

Comment:
Should `unknown[]` behere, considering your comment in the description?

> There is one functional change here, which is that the streams will ignore `MessageEvent.data` values for which `Array.isArray()` returns true.

It seems like the code is disallowing an array, but the type is allowing it.

Reply:
Ah, this `data` refers to `PostMessageEvent.data.data`, which can be an array, not `PostMessageEvent.data`, which cannot. 😅

Reply:
The property names here are unfortunate, but this blow was unfortunately struck years before the present date. 🎻

Reply:
Cool, makes sense!